### ISSUE TITLE
How to extract text based on a defined region by the user?

### ISSUE THREAD

**Issue by RitchieP** [pdfminer/pdfminer.six#878]
I'm currently working on a project that requires me to extract text on a region of a page defined by the user. The lack of documentation around the code is really making this challenging. I would hope some experts in this library could help guide me through this.

**Comment by QuentinAndre11** [comment]
I suggest you to look at PDFPlumber, built on pdfminer.six, which is actively maintained and well documented

**Comment by RitchieP** [comment]
Thanks for the suggestion! I also came across a library called PyMuPDF. Would like to hear thoughts about this library also. And how does it compare with PDFPlumber.

**Comment by krishnasism** [comment]
We are working on a pdf extraction tool called hotpdf with full MIT license. It's written on top of pdfminer.six :)

https://github.com/weareprestatech/hotpdf

**Comment by QuentinAndre11** [comment]
@krishnasism I'm actually interested in a memory optimized tool built for larger PDFs on top of pdfminer.six. I have a question tho: I saw that you can extract texts based on a bounding box and you use a memory map to do so. Is it spatially indexed or is it a simple loop like PDFPlumber ?

**Comment by krishnasism** [comment]
It uses an indexed sparse matrix. While retrieving it loops in range of query

**Comment by QuentinAndre11** [comment]
Hm, unfortunately it doesn't fit my use case (I also need to retrieve the curves), but that's interesting

**Comment by RitchieP** [comment]
Been quite some time since I last visited this page and the project was long ago. Thanks @krishnasism for bringing this back up and such an awesome library!

I thought I might as well provide the solution to this issue I originally opened. (I'm sorry if my solution is a bit wrong or outdated since I finished the project long ago)

The steps are below:

1. I believe I used PyPDF (or some other PDF library) to convert the PDF page into an image. 
2. Pass that image to OpenCV, and draw regions with it. (In my case, it was just rectangles) 
3. After I've drawn the rectangles, I could get their coordinates.
4. Coordinates are then passed back to `pdfminer.six`, and there is where I can then extract text based on a region.
